### ISSUE TITLE
systemd initramfs service missing dependencies for the key device

### ISSUE THREAD

**Issue by grawity** [electrickite/mkinitcpio-tpm2-encrypt#7]
I store the TPM-sealed key file in the EFI system partition¹ and boot with:

    tpmkey=/dev/disk/by-partuuid/c50b4d4f-8eb4-4f47-a48c-5a521e8b7666:EFI/Linux/system.tpm:0x81000001

This doesn't seem to work with systemd-based initramfs:

```
Dec 28 22:35:26 archlinux systemd[1]: Reached target Sockets.
Dec 28 22:35:26 archlinux systemd[1]: Starting Create list of static device nodes for the current kernel...
Dec 28 22:35:26 archlinux systemd[1]: Starting Journal Service...
Dec 28 22:35:26 archlinux systemd[1]: Starting Load Kernel Modules...
Dec 28 22:35:26 archlinux systemd[1]: Starting Coldplug All udev Devices...
Dec 28 22:35:26 archlinux systemd[1]: Starting Unseal key from tpm2...
Dec 28 22:35:26 archlinux systemd[1]: Finished Create list of static device nodes for the current kernel.
Dec 28 22:35:26 archlinux systemd[1]: Starting Create Static Device Nodes in /dev...
Dec 28 22:35:26 archlinux systemd[1]: Finished Load Kernel Modules.
Dec 28 22:35:26 archlinux kernel: tpm_tis MSFT0101:00: 2.0 TPM (device-id 0x1B, rev-id 16)
Dec 28 22:35:26 archlinux systemd[1]: Finished Create Static Device Nodes in /dev.
Dec 28 22:35:26 archlinux systemd[1]: Starting Rule-based Manager for Device Events and Files...
Dec 28 22:35:26 archlinux systemd[1]: Started Rule-based Manager for Device Events and Files.
Dec 28 22:35:26 archlinux systemd[1]: tpm2-unseal.service: Main process exited, code=exited, status=127/n/a
Dec 28 22:35:26 archlinux systemd[1]: tpm2-unseal.service: Failed with result 'exit-code'.
Dec 28 22:35:26 archlinux systemd[1]: Failed to start Unseal key from tpm2.
Dec 28 22:35:26 archlinux kernel: fbcon: Taking over console
Dec 28 22:35:26 archlinux systemd[1]: Reached target Local Encrypted Volumes (Pre).
Dec 28 22:35:26 archlinux kernel: Console: switching to colour frame buffer device 240x67
Dec 28 22:35:26 archlinux systemd-journald[124]: Journal started
Dec 28 22:35:26 archlinux systemd-journald[124]: Runtime Journal (/run/log/journal/d27c9518021e450d83fec7c4afa646a5) is 8.0M, max 156.7M, 148.7M free.
Dec 28 22:35:26 archlinux systemd-modules-load[125]: Inserted module 'vfat'
Dec 28 22:35:26 archlinux systemd-modules-load[125]: Module 'nls_cp437' is built in
Dec 28 22:35:26 archlinux systemd-modules-load[125]: Inserted module 'nls_iso8859_1'
Dec 28 22:35:26 archlinux systemd-udevd[134]: Network interface NamePolicy= disabled on kernel command line, ignoring.
Dec 28 22:35:26 archlinux tpm2_encrypt_hook[149]: mount: /tpmkey: special device /dev/disk/by-partuuid/c50b4d4f-8eb4-4f47-a48c-5a521e8b7666 does not exist.
Dec 28 22:35:26 archlinux tpm2_encrypt_hook[154]: umount: /tpmkey: not mounted.
Dec 28 22:35:26 archlinux tpm2_encrypt_hook[127]: TPM keyfiles could not be opened
Dec 28 22:35:26 archlinux systemd[1]: Started Journal Service.
Dec 28 22:35:26 archlinux tpm2_encrypt_hook[157]: /usr/bin/tpm2_encrypt_hook: line 7: /usr/bin/systemd-cat: not found
Dec 28 22:35:26 archlinux tpm2_encrypt_hook[156]: sh: write error: Broken pipe
Dec 28 22:35:26 archlinux systemd[1]: Finished Coldplug All udev Devices.
```

¹ (I'm hesitant to use TPM NVRAM as I haven't been able to find any information about its write endurance.)

**Comment by electrickite** [comment]
@grawity Were you able to resolve the issue with the device node?

**Comment by grawity** [comment]
...I don't remember if I eventually found a workaround or not. But I think I had to hardcode an `After=` for the specific device that it needs to mount, as a temporary hack.

Eventually, though, I switched to using systemd's newly built-in `systemd-cryptenroll` (which stores the tpmkey directly in LUKS2 metadata, so the only dependency ). I still have a custom script to produce and store the tpmkey in the format it expects, but the unlocking is automatic with just the `sd-encrypt` hook.
